command/job_run.go

```diff
@@ -101,6 +101,11 @@ Run Options:
     the job file. This overrides the token found in $VAULT_TOKEN environment
     variable and that found in the job.
 
+  -vault-namespace
+    If set, the passed Vault namespace is stored in the job before sending to the
+    Nomad servers. This overrides the namespace found in $VAULT_NAMESPACE environment
```

Comment:
I just realized this is a bit of a backwards incompatibility:

Before Nomad ignored this environment variable and would have used the agent-configured Vault namespace.

Now the operator's environment will override the agent-configuration.

I could see this being an issue for CI/CD environments which may use a distinct Vault namespace from the app being deployed.

However, VAULT_TOKEN is already prior art for `nomad run`'s environment to be *app* specific.

To be safe, let's call out this change in the Version Specific Upgrade docs.